bigquery_etl/cli/metadata.py

```diff
@@ -125,4 +127,49 @@ def publish(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> Non
         except FileNotFoundError:
             print("No metadata file for: {}.{}.{}".format(project, dataset, table))
 
+
+@metadata.command(
+    help="""
+    Deprecate BigQuery table by updating metadata.yaml file.
+    Deletion date is by default 3 months from current date if not provided.
+
+    Example:
+     ./bqetl metadata deprecate ga_derived.downloads_with_attribution_v2 --deletion_date=2024-03-02
+    """
+)
+@click.argument("name")
+@project_id_option(
+    ConfigLoader.get("default", "project", fallback="moz-fx-data-shared-prod")
+)
+@sql_dir_option
+@click.option(
+    "--deletion_date",
+    "--deletion-date",
+    help="Date when table is scheduled for deletion. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    default=datetime.today() + relativedelta(months=+3),
+)
+def deprecate(
+    name: str,
+    sql_dir: str,
+    project_id: str,
+    deletion_date: datetime,
+) -> None:
+    """Deprecate Bigquery table by updating metadata yaml file(s)."""
+    table_metadata_files = paths_matching_name_pattern(
+        name, sql_dir, project_id=project_id, files=["metadata.yaml"]
+    )
+
+    for metadata_file in table_metadata_files:
+        metadata = Metadata.from_file(metadata_file)
+
+        metadata.deprecated = True
+        metadata.deletion_date = deletion_date.date()
+
+        metadata.write(metadata_file)
+        click.echo(f"Updated {metadata_file} with deprecation.")
+
+    if not table_metadata_files:
+        raise FileNotFoundError(f"No metadata file(s) were found for: {name}")
+
     return None
```

Comment:
I just noticed this other `return None` that could be removed, and that actually this is currently resulting in `return None` being removed from the `publish()` function, which is unrelated to this PR.

So I'd suggest removing `return None` here and putting it back at the end of the `publish()` function to avoid having an unrelated change.